flux-typeck/src/checker.rs

```diff
@@ -379,8 +422,9 @@ impl<'a, 'tcx, P: Phase> Checker<'a, 'tcx, P> {
         &mut self,
         rcx: &mut RefineCtxt,
         env: &mut TypeEnv,
+        tag: Tag,
```

Comment:
I'd pass the `source_info` as an argument here and create the tag inside `check_ret`